skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,192 @@
+    >>> result = rolling_ellipsoid(data.coins())
+    """
+
+    kernel_size = np.asarray(kernel_size)
+    if not np.issubdtype(kernel_size.dtype, np.number):
+        raise ValueError(
+            "kernel_size must be convertible to a numeric array.")
+    if not kernel_size.shape == (2,):
+        raise ValueError(
+            "kernel_size must be a two element tuple.")
+    if np.any(kernel_size <= 0):
+        raise ValueError("All elements of kernel_size must be greater zero.")
+    kernel_size = kernel_size / 2
+
+    intensity_vertex = np.asarray(intensity_vertex, dtype=np.float_)
+    if not intensity_vertex.shape == tuple():
+        raise ValueError("Intensity_vertex must be a scalar.")
+    if np.any(intensity_vertex <= 0):
+        raise ValueError("Intensity_vertex must be greater zero.")
+    intensity_vertex = intensity_vertex / 2
+
+    image = np.asarray(image)
+    img = image.astype(np.float_)
+
+    kernel_size_y, kernel_size_x = np.round(kernel_size).astype(int)
+
+    pad_amount = [(0, 0)] * image.ndim
+    pad_amount[-2] = (kernel_size_x, kernel_size_x)
+    pad_amount[-1] = (kernel_size_y, kernel_size_y)
+    img = np.pad(img, pad_amount,  constant_values=np.Inf, mode="constant")
+
+    tmp_x = np.arange(-kernel_size_x, kernel_size_x + 1)
+    tmp_y = np.arange(-kernel_size_y, kernel_size_y + 1)
+    x, y = np.meshgrid(tmp_x, tmp_y)
+
+    kernel_size_y, kernel_size_x = kernel_size
+    tmp = (x / kernel_size_x) ** 2 + (y / kernel_size_y) ** 2
+    cap_height = intensity_vertex - intensity_vertex * \
+        np.sqrt(np.clip(1 - tmp, 0, None))
+    cap_height = cap_height.astype(np.float_)
+
+    kernel = np.asarray(tmp <= 1, dtype=np.float_)
+    kernel[kernel == 0] = np.Inf
+
+    if has_nan:
+        strides = (img.itemsize, img.strides[-2], img.itemsize)
+        shape = (img.size - (kernel.shape[0] - 1) * img.shape[-1] -
+                 (kernel.shape[1] - 1), *kernel.shape)
+        windowed = as_strided(img, shape, strides)
+        background = apply_kernel_nan(windowed, kernel, cap_height)
+        background = as_strided(background, image.shape, img.strides)
+    else:
+        # windowed = view_as_windows(img, kernel.shape)
+        # background = apply_kernel(windowed, kernel, cap_height)
+        strides = (img.itemsize, img.strides[-2], img.itemsize)
+        shape = (img.size - (kernel.shape[0] - 1) * img.shape[-1] -
+                 (kernel.shape[1] - 1), *kernel.shape)
+        windowed = as_strided(img, shape, strides)
+        background = apply_kernel_flat(windowed, kernel, cap_height)
+        background = as_strided(background, image.shape, img.strides)
+
+    background = background.astype(image.dtype)
+
+    filtered_image = image - background
+
+    return filtered_image
+
+
+def rolling_ball(image, radius=50, has_nan=False):
+    """
+    Estimate background intensity using a rolling ball.
+
+    The rolling ball algorithm estimates background intensity of a grayscale
```

Comment:
Maybe refer to the `rolling_ellipsoid` docstring instead of duplicating that much content?